lib/puma/cluster.rb

```diff
@@ -322,6 +343,34 @@ def setup_signals
               w.last_status[:requests_count] >= fork_requests
           end
         end
+
+      end
+      if @options[:mold_worker]
+        Signal.trap "SIGURG" do
+          mold_and_refork!
```

Comment:
on SIGURG, promote a new mold (whoever's got the highest request count) and replace all workers with reforks